internal/graph/cycle_check_resolver_test.go

```diff
@@ -0,0 +1,62 @@
+package graph
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+	"go.uber.org/mock/gomock"
+
+	"github.com/openfga/openfga/pkg/tuple"
+)
+
+func TestCycleDetectionCheckResolver(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+
+	ctx := context.Background()
+
+	cycleDetectionCheckResolver := NewCycleDetectionCheckResolver()
+	t.Cleanup(cycleDetectionCheckResolver.Close)
+
+	t.Run("detects_cycle_and_returns_cycle_detected_error", func(t *testing.T) {
+		cyclicalTuple := tuple.NewTupleKey("document:1", "viewer", "user:will")
+
+		visitedPaths := make(map[string]struct{}, 0)
+		visitedPaths[tuple.TupleKeyToString(cyclicalTuple)] = struct{}{}
+
+		resp, err := cycleDetectionCheckResolver.ResolveCheck(ctx, &ResolveCheckRequest{
+			StoreID:            uuid.NewString(),
+			TupleKey:           cyclicalTuple,
+			ResolutionMetadata: &ResolutionMetadata{Depth: defaultResolveNodeLimit},
+			VisitedPaths:       visitedPaths,
+		})
+
+		require.ErrorIs(t, err, ErrCycleDetected)
+		require.False(t, resp.GetAllowed())
+	})
+
+	t.Run("correctly_undetects_cycle_and_returns_no_error", func(t *testing.T) {
```

Comment:
This name is weird to me as "undetects" implies we're somehow doing inverted logic. Can we just rename to simply `no_cycle_returns_no_error`?